app/src/main/cpp/ExternalVR.cpp

```diff
@@ -283,12 +293,24 @@ ExternalVR::PushSystemState() {
 
 void
 ExternalVR::PullBrowserState() {
-  Lock lock(m.data.browserMutex);
+
+  pthread_mutex_t* browserMutex;
+  if (m.sourceBrowser == ExternalVR::VRBrowserType::Gecko) {
+    browserMutex = &m.data.geckoMutex;
+  } else {
+    browserMutex = &m.data.servoMutex;
+  }
+  Lock lock(*browserMutex);
   if (lock.IsLocked()) {
    m.PullBrowserStateWhileLocked();
   }
 }
 
+void ExternalVR::SetSourceBrowser(VRBrowserType aBrowser) {
+  m.Reset();
+  m.sourceBrowser = aBrowser;
```

Comment:
Do the if here and set the mutex & cond references according to the browser type